src/nvim/ex_session.c

```diff
@@ -1013,9 +1013,8 @@ void ex_mkrc(exarg_T *eap)
           }
           shorten_fnames(true);
           // restore original dir
-          if (*dirnow != NUL && ((ssop_flags & SSOP_SESDIR)
-                                 || ((ssop_flags & SSOP_CURDIR) && globaldir !=
-                                     NULL))) {
+          if ((ssop_flags & SSOP_SESDIR)
+              || ((ssop_flags & SSOP_CURDIR) && globaldir != NULL)) {
```

Comment:
I think this whole `if`-block should be removed. It likely comes from a messed-up rebase in <https://github.com/neovim/neovim/commit/f9d4c78ddcd99efeeb4e4e44b7bd0b6e34889db3>.